frontend/src/metabase/nav/containers/MainNavbar/SidebarItems/DwhUploadCSV/DwhUploadCSV.tsx

```diff
@@ -0,0 +1,94 @@
+import { type ChangeEvent, useCallback, useRef, useState } from "react";
+import { t } from "ttag";
+
+import {
+  type CollectionOrTableIdProps,
+  ModelUploadModal,
+} from "metabase/collections/components/ModelUploadModal";
+import type { OnFileUpload } from "metabase/collections/types";
+import { UploadInput } from "metabase/components/upload";
+import { useToggle } from "metabase/hooks/use-toggle";
+import { useDispatch } from "metabase/lib/redux";
+import {
+  type UploadFileProps,
+  uploadFile as uploadFileAction,
+} from "metabase/redux/uploads";
+import { Box, Button, Icon } from "metabase/ui";
+
+import { trackDWHUploadCSVClicked } from "./analytics";
+
+export const DwhUploadCSV = () => {
+  const [
+    isModelUploadModalOpen,
+    { turnOn: openModelUploadModal, turnOff: closeModelUploadModal },
+  ] = useToggle(false);
+
+  const [uploadedFile, setUploadedFile] = useState<File | null>(null);
+
+  const dispatch = useDispatch();
+  const uploadFile = useCallback(
+    ({ file, modelId, collectionId, tableId, uploadMode }: UploadFileProps) =>
+      dispatch(
+        uploadFileAction({ file, modelId, collectionId, tableId, uploadMode }),
+      ),
+    [dispatch],
+  );
```

Comment:
nit: for simplicity maybe there's no need to destructure the props? But maybe it's more explicit to do so

Reply:
https://github.com/metabase/metabase/blob/master/frontend/src/metabase/collections/components/CollectionContent/CollectionContent.tsx#L68